### PR TITLE
Fix: Update Reducers

### PR DESCRIPTION
**Pull request by hyperxtend**

### CODE REVIEW

File: src/reducers/status/reducers.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        ...state,
+        history: state.history.slice(0, action.step + 1),
+        moveNumber: action.step,
+        xIsNext: !(action.step % 2),
+      };
+
+    case SELECT_SQUARE:
+      if (calculateWinner(squares) || squares[action.index]) {
+        return state;
+      }
```
Comment (strange-developer) [inline src/reducers/status/reducers.js]: You can move this functionality further up. You don't need to dispatch an action if the `if` statement criteria is met.

File: src/reducers/status/reducers.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      squares[action.index] = state.xIsNext ? 'X' : 'O';
```
Comment (strange-developer) [inline src/reducers/status/reducers.js]: This calculation can also be done higher in the process before calling the action. Your payload can consist of the next player name. Try to keep all calculations out of the reducer as far as possible.

**Review by strange-developer — CHANGES_REQUESTED** [review]

File: src/reducers/status/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from './reducers';
```
Comment (strange-developer) [inline src/reducers/status/index.js]: The folder name `status`. What does this refer to? Perhaps change the name to be more descriptive.

**Review by strange-developer — CHANGES_REQUESTED** [review]

File: src/reducers/status/reducers.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { calculateWinner } from '../../components/game/controller';
```
Comment (strange-developer) [inline src/reducers/status/reducers.js]: Call this file: `reducer.js` as opposed to `reducers.js`. Only one reducer currently exists in this file.

**Review by strange-developer — CHANGES_REQUESTED** [review]

File: src/components/game/container.js [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
+  history: history.app.status.history,
+  moveNumber: history.app.status.moveNumber,
+  xIsNext: history.app.status.xIsNext,
+  state: history.app.status,
 });
 
 const mapDispatchToProps = (dispatch) => ({
-  onSelectSquare: (id) => dispatch(selectSquare(id)),
+  onSelectSquare: (squareId, xIsNext) => {
+    const squaresPlayed = gameState(mapStateToProps.history);
```
Comment (strange-developer) [inline src/components/game/container.js]: ```gameState(mapStateToProps.history);``` 
What does this piece do?

File: src/components/game/controller.js [hunk 6]
```diff
@@ -39,3 +41,13 @@ export const determineGameStatus = (winner, moveNumber, xIsNext) => {
   }
   return `Next player is ${xIsNext ? 'X' : 'O'}`;
 };
+
+export const gameState = (state = initialState) => {
```
Comment (strange-developer) [inline src/components/game/controller.js]: What is state here?

**Review by strange-developer — CHANGES_REQUESTED** [review]

File: src/components/game/container.js (hunk 5 above)
Comment (hyperxtend) [reply src/components/game/container.js]: I was trying to pass the history to the function but it wasn't working so removed it

File: src/components/game/controller.js (hunk 6 above)
Comment (hyperxtend) [reply src/components/game/controller.js]: The state is the initial state of the game

File: src/components/game/container.js [hunk 7]
```diff
@@ -1,20 +1,36 @@
...
+  onSelectSquare: (squareIndex, nextPlayer, history, moveNumber) => {
+    const currentMovesPlayed = getCurrentMovesPlayed(
+      history,
+      moveNumber,
+      squareIndex,
+      nextPlayer
+    );
+    if (!currentMovesPlayed[squareIndex]) {
+      currentMovesPlayed[squareIndex] = nextPlayer ? 'X' : 'O';
+      dispatch(selectSquare({ squareIndex, currentMovesPlayed }));
```
Comment (strange-developer) [inline src/components/game/container.js:30]: Add in a test that makes sure `selectSquare` gets called with the necessary parameters

File: src/reducers/state-of-game/reducer.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+export const initialState = {
+  history: [Array(9).fill('')],
+  xIsNext: true,
+  moveNumber: 0,
+};
+
+export default (state = initialState, action = {}) => {
+  switch (action.type) {
+    case GO_TO_MOVE:
```
Comment (strange-developer) [inline src/reducers/state-of-game/reducer.js:11]: Add tests for each action in the reducer.

**Review by strange-developer — CHANGES_REQUESTED** [review]

File: src/components/game/container.js (hunk 7 above)
Comment (hyperxtend) [reply src/components/game/container.js:30]: Will do

Comment (strange-developer) [reply src/components/game/container.js:30]: `selectSquare` still needs to be tested to ensure that it's being called with the correct parameters.

**Review by strange-developer — APPROVED** [review]

**Pull request merged**